pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -54,7 +62,8 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 	// Only deploy when the version has advanced past 0.
 	if config.LatestVersion == 0 {
 		glog.V(5).Infof("Waiting for first version of %s", deployutil.LabelForDeploymentConfig(config))
-		return nil
+		// Check for any possible failures anyway
+		return c.failureInspection(config, nil)
```

Comment:
Would it work to just do the message update right away before doing anything else in this function? I'm not sure there's any reason to do it conditionally throughout.